src/util/AttachmentTools.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  Attachment,
+  MeshAttachment,
+  RegionAttachment,
+  TextureAtlasRegion,
+} from "@esotericsoftware/spine-core";
+
+/**
+ * Creates a new `RegionAttachment` from a specified texture atlas region.
+ *
+ * @param region - The texture atlas region used to create the attachment.
+ * @param attachmentName - The name of the new attachment.
+ * @param scale - A scaling factor applied to the attachment's dimensions (default is 1).
+ * @param rotation - The rotation angle of the attachment in degrees (default is 0).
+ * @returns The created `RegionAttachment` with the passed in region.
+ */
+export function createAttachmentFromRegion(
+  region: TextureAtlasRegion,
+  attachmentName: string,
+  scale: number = 1,
+  rotation: number = 0
+): RegionAttachment {
+  const attachment = new RegionAttachment(attachmentName, region.name);
+  attachment.region = region;
+  attachment.scaleX = 1;
+  attachment.scaleY = 1;
+  attachment.rotation = rotation;
+
+  const originalWidth = region.originalWidth;
+  const originalHeight = region.originalHeight;
+  attachment.width = originalWidth * scale;
```

Comment:
originlWidth and originalHeight both used once, not need cache

Reply:
done